src/python/pants/bin/exception_sink.py

```diff
@@ -0,0 +1,43 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import datetime
+from builtins import object
+
+from pants.util.dirutil import is_writable_dir
+from pants.util.memo import memoized_classproperty
+
+
+class ExceptionSink(object):
+  """A mutable singleton object representing where exceptions should be logged to."""
+
+  @memoized_classproperty
+  def instance(cls):
+    return cls()
+
+  def __init__(self):
+    self._cached_destination = None
+
+  class ExceptionSinkError(Exception): pass
+
+  def _get_timestamp(self):
```

Comment:
Rather than adding timestamp logic, I'd recommend moving the `Exiter._log_exception` method into this class.

And I think that if you guarantee that `self._cached_destination` is always "something" by initializing it to `cwd`, then you should be able to have `set_destination` log to the previous destination if it fails.

Reply:
I have done that and added thorough testing in e562a49!